api/controller/classes.py

```diff
@@ -145,6 +175,53 @@ def fetch_geom_from_db(dataset, geom_id):
    return o
 
 
+def find_geometry_by_latlng(latlng, dataset=None, crs='4326'):
+   """
+   Assumes there is a Postgis database with connection config specified in system environment variables.
+   Also assumes there is a table/view called 'combined_geoms' with structure (id, dataset, geom).
+   This function connects to the DB, and queries for matching geoms based on input latlng parameters.
+   Default CRS is WGS84 (4326)
+   """
+   if latlng is None or not("," in latlng):
+     return { 'count': -1, 'res': None, 'errcode': 1}
+   arrData = latlng.split(',')
+   db_name = os.environ['GSDB_DBNAME']
+   db_host = os.environ['GSDB_HOSTNAME']
+   db_port = os.environ['GSDB_PORT']
+   username = os.environ['GSDB_USER']
+   passwd = os.environ['GSDB_PASS']
+   conn = psycopg2.connect(dbname=db_name, host=db_host, port=db_port, user=username, password=passwd)
+   cur = conn.cursor()
+   query = 'select id, dataset from combined_geoms where ST_Intersects( ST_Transform(ST_SetSRID(ST_Point({para1}, {para2}),{crs}),3577) , geom);'.format(para1=arrData[0], para2=arrData[1], crs=crs)

```

Comment:
I know we're not very concerned with security in these applications, but in the name of best practice, can you please change this to use native Postgres Query Parameters, rather than string.format().
https://www.psycopg.org/docs/usage.html#the-problem-with-the-query-parameters
See:
> Warning
>
>Never, never, NEVER use Python string concatenation (+) or string parameters interpolation (%) to pass variables to a SQL query string. Not even at gunpoint.

Reply:
like this:
```
query = 'SELECT id, dataset FROM combined_geoms WHERE ST_Intersects( ST_Transform(ST_SetSRID(ST_Point(%s, %s), %s),3577) , geom);'
...
cur.execute(query, (arrData[0], arrData[1], crs)) 
```
Note the input variables are never mixed into to the query string, but passed along to the postgres execute function.

Reply:
Thanks. Fixed that one now.